### PR TITLE
Reduce flakiness on E2E setup

### PR DESCRIPTION
**Pull request by Luc45**
Reduce flakiness on E2E test setup.

When waiting for `networkidle`, it would try to find the text in the selector too fast, which would fail and cause the admin login to repeat. However, when repeating, it was already logged in, which would cause it to fail all retries, as it couldn't find the login form anymore.

This PR fixes this behavior by waiting for `domcontentload` instead, so that we check the text in the selector after the DOM is fully loaded, which seems to have fixed the flakiness we have observed in QIT E2E tests.

### CODE REVIEW

**Comment by Luc45** [comment]
Actually, that's not it. After authenticating, it seems the user is redirected to WooCommerce Setup:

First try - Before Login:

image

First try - After login:

image

Second, third, fourth and fifth try - Before/after login, trying to find login form and failing:

image

So I'll add a `adminPage.goto('/wp-admin');` as it was before this commit https://github.com/woocommerce/woocommerce/commit/64320a2c5cead883181eb0f35735b046b9531fc0#diff-d942b6f3fa3b15ec2af416649622ebf27e96e44dc5040cea6a22bfcd81b87b89R68-R71, to make sure we are in admin dashboard after login.

**Comment by Luc45** [comment]
That first redirect seems to come from the first hit to `wp-admin` after activating WooCommerce, here: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingSetupWizard.php#L70-L74

File: plugins/woocommerce/tests/e2e-pw/global-setup.js [hunk 1]
```diff
@@ -60,6 +60,8 @@ module.exports = async ( config ) => {
 			await adminPage.fill( 'input[name="pwd"]', admin.password );
 			await adminPage.click( 'text=Log In' );
 			await adminPage.waitForLoadState( 'networkidle' );
+			await adminPage.goto( `/wp-admin` );
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/global-setup.js:63]: Makes sense to me.  Better to not leave things up to chance, so this is a great addition.

**Review by lanej0 — APPROVED** [review]
Looks good!

File: plugins/woocommerce/tests/e2e-pw/global-setup.js (hunk 1 above)
Comment (Luc45) [reply plugins/woocommerce/tests/e2e-pw/global-setup.js:63]: Yup, I just found out where the redirect came from, I've shared some details on a comment I've posted 1 min ago, not sure if you had time to see it yet. We could either go to wp-admin, or we could disable the onboarding wizard.

**Comment by Luc45** [comment]
Thanks a lot for the quick review here, @lanej0 !

**Pull request merged**